packages/authentication/src/metadata-provider.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingKeys} from '@loopback/core';
+import {Constructor, Provider, inject} from '@loopback/context';
+import {AuthenticationMetadata, getAuthenticateMetadata} from './decorator';
+
+/**
+ * @description Provides authentication metadata of a controller method
+ * @example `context.bind('authentication.meta')
+ *   .toProvider(AuthMetadataProvider)`
+ */
+export class AuthMetadataProvider<T>
```

Comment:
I don't think AuthMetadata has anything to do with the controller type. There is no need to use `<T>`.